tests/test_preprocessing.py

```diff
@@ -0,0 +1,605 @@
+        dummy_sector1_dir / "dummy_sector1.zarr", consolidated=True, mode="w"
+    )
+
+    damages_ds_2 = (
+        xr.Dataset(
+            {
+                "delta_dummy2": (
+                    ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                    np.ones((2, 2, 2, 2, 2, 2, 2)),
+                ),
+                "histclim_dummy2": (
+                    ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                    np.ones((2, 2, 2, 2, 2, 2, 2)),
+                ),
+            },
+            coords={
+                "gcm": (["gcm"], ["ABCD1", "surrogate_GCM"]),
+                "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+                "rcp": (["rcp"], ["rcp45", "rcp85"]),
+                "ssp": (["ssp"], ["SSP3", "SSP4"]),
+                "batch": (["batch"], [0, 1]),
+                "region": (["region"], ["ZWE.test_region", "USA.test_region"]),
+                "year": (["year"], [2022, 2023]),
+            },
+        )
+        + 1
+    )
+
+    damages_ds_2.to_zarr(
+        dummy_sector2_dir / "dummy_sector2.zarr", consolidated=True, mode="w"
+    )
+
+    damages_out_expected = (
+        (
+            xr.Dataset(
+                {
+                    "summed_delta": (
+                        ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                        np.ones((2, 2, 2, 2, 2, 2, 2)),
+                    ),
+                    "summed_histclim": (
+                        ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                        np.ones((2, 2, 2, 2, 2, 2, 2)),
+                    ),
+                },
+                coords={
+                    "gcm": (["gcm"], ["ABCD1", "surrogate_GCM"]),
+                    "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+                    "rcp": (["rcp"], ["rcp45", "rcp85"]),
+                    "ssp": (["ssp"], ["SSP3", "SSP4"]),
+                    "batch": (["batch"], [0, 1]),
+                    "region": (["region"], ["ZWE.test_region", "USA.test_region"]),
+                    "year": (["year"], [2022, 2023]),
+                },
+            )
+            + 2
+        )
+        .astype(np.float32)
+        .chunk(
+            {
+                "batch": 15,
+                "ssp": 1,
+                "model": 1,
+                "rcp": 1,
+                "gcm": 1,
+                "year": 10,
+                "region": 24378,
+            }
+        )
+    )
+
+    sum_AMEL(sectors, config_in, "dummy_AMEL")
+    xr.testing.assert_equal(
+        xr.open_zarr(dummy_AMEL_dir / "dummy_AMEL.zarr"), damages_out_expected
+    )
+
+
+@pytest.mark.parametrize(
+    "recipe, eta, batchsize",
+    [
+        ("adding_up", 10, 15),
+        ("adding_up", None, 15),
+        ("adding_up", None, 5),
+        ("risk_aversion", 10, 15),
+    ],
+)
+def test_reduce_damages(tmp_path, recipe, eta, batchsize):
+    if recipe == "adding_up" and eta is not None:
```

Comment:
The if/elses here hint this test is testing more than one behavior. I'd consider breaking it down into multiple tests so that each test is checking for one behavior or has one reason to fail. So, in this case, maybe make one test function that checks that this error gets raised, and then another test function checking "the happy path" - testing the functions spits out the correct output, etc. It becomes harder and harder to reason why a test failed or what a test is actually testing if a test starts checking for too many different things.

There are a couple of other places that might have this issue. Happy to discuss this more if needed.